utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -487,6 +487,40 @@ Status WriteBatchWithIndex::GetFromBatch(ColumnFamilyHandle* column_family,
   return s;
 }
 
+Status WriteBatchWithIndex::GetEntityFromBatch(
+    ColumnFamilyHandle* column_family, const Slice& key,
+    PinnableWideColumns* columns) {
+  if (!column_family) {
+    return Status::InvalidArgument(
+        "Cannot call GetEntityFromBatch without a column family handle");
+  }
+
+  if (!columns) {
+    return Status::InvalidArgument(
+        "Cannot call GetEntityFromBatch without a PinnableWideColumns object");
+  }
+
+  MergeContext merge_context;
+  Status s;
+  auto result = WriteBatchWithIndexInternal::GetEntityFromBatch(
+      this, column_family, key, &merge_context, columns, &s);
+
+  if (result == WBWIIteratorImpl::kDeleted ||
```

Comment:
nit: Do you think `switch` might be better here? (`GetFromBatch()` seems to be using `switch` and I see what we are doing there is identical)

Reply:
Personally, I feel this flows better - in `GetFromBatch`, two `case`s (`kError` and `kFound`) somewhat unintuitively `break` out immediately, which necessitates a comment about why this is done. There's also a separate `default` there that does nothing but `assert(false);`; here, the assertion on line 519 ensures we cover all possible enum values.

Reply:
This flow sounds good to me as well. I just feel that it would be better if we don't have two places doing the same thing.

Maybe `GetStatusFromWBWIIterResult()` can be introduced, but not a big deal.